File: src/PlasticisingTile.API/DTO/Plasticising/PlasticisingTileConfigureRequestDto.cs
using PlasticisingTile.API.DTO.Interfaces;
using PlasticisingTile.API.DTO.Shared;
using PlasticisingTile.Core.Enums;

namespace PlasticisingTile.API.DTO.Plasticising;

public class PlasticisingTileConfigureRequestDto : ITileRequestDto
{
    public DateTimeRangeFilterDto? DateTimeRangeFilter { get; set; }
    public IEnumerable<string> SelectedColumnKeys { get; set; } = new List<string>();
    public IEnumerable<PlasticisingTileAggregationEnum> SelectedAggregations { get; set; } = new List<PlasticisingTileAggregationEnum>();
}

File: src/plasticising-tile-webapp/src/app/api/models/plasticising-tile-configure-request-dto.ts
/* tslint:disable */
/* eslint-disable */
import { DateTimeRangeFilterDto } from './date-time-range-filter-dto';
import { PlasticisingTileAggregationEnum } from './plasticising-tile-aggregation-enum';
export interface PlasticisingTileConfigureRequestDto {
  dateTimeRangeFilter?: DateTimeRangeFilterDto;
  selectedAggregations?: null | Array<PlasticisingTileAggregationEnum>;
  selectedColumnKeys?: null | Array<string>;
}
